persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedActorOf.scala

```diff
@@ -0,0 +1,135 @@
+package com.evolutiongaming.akkaeffect.persistence
+
+import akka.actor.Actor
+import akka.persistence.SnapshotSelectionCriteria
+
+import cats.effect.implicits.effectResourceOps
+import cats.effect.{Async, Resource, Ref}
+import cats.syntax.all._
+
+import com.evolutiongaming.akkaeffect._
+import com.evolutiongaming.catshelper.ToFuture
+
+import scala.reflect.ClassTag
+import java.time.Instant
+
+object EventSourcedActorOf {
+
+  /** Describes lifecycle of entity with regards to event sourcing & PersistentActor Lifecycle phases:
+    *
+    *   1. RecoveryStarted: we have id in place and can decide whether we should continue with recovery 
+    *   2. Recovering : reading snapshot and replaying events 
+    *   3. Receiving : receiving commands and potentially storing events & snapshots 
+    *   4. Termination : triggers all release hooks of allocated resources within previous phases
+    *
+    * @tparam S snapshot
+    * @tparam E event
+    * @tparam C command
+    */
+  type Lifecycle[F[_], S, E, C] =
+    Resource[F, RecoveryStarted[F, S, E, Receive[F, Envelope[C], ActorOf.Stop]]]
```

Comment:
The comment is a bit confusing, at least to me as it says there could be different lifecycle phases, but it contains `RecoveryStarted[...]` inside of the signature.

May be I will understand better, though, when I read all the code. I decided to mention it, as it may also be confusing to other readers.